api/nodes.go

```diff
@@ -215,6 +224,7 @@ func (n *Nodes) monitorDrainNode(ctx context.Context, nodeID string, index uint6
 				msg = Messagef(MonitorMsgLevelInfo, "Node %q has marked all allocations for migration", nodeID)
 			} else {
 				msg = Messagef(MonitorMsgLevelInfo, "No drain strategy set for node %s", nodeID)
+				defer cancel()
```

Comment:
Why here and not before the if statement?

Reply:
@nickethier So by cancelling the shared context we can cancel the alloc watcher goroutine as well. We only want to do that if we have no expectation of allocations ever changing state because of a drain. That is only true in the else statement because the node has no drain strategy set and it never did. In the other case, allocations can still be transitioning.

Reply:
Ah yes I get it now, thanks!